From 0011ddf426474630849c8628d28e690fcbcdb7da Mon Sep 17 00:00:00 2001
From: Tad <tad@spotco.us>
Date: Tue, 1 Dec 2020 03:57:45 -0500
Subject: [PATCH] mwlwifi: Disable tx_amsdu

This fixes the high latencies.
Unsure who figured this out.
https://forum.archive.openwrt.org/viewtopic.php?id=64949&p=161
https://forum.openwrt.org/t/linksys-wrt1200ac-wireless-performance-on-latest-stable/39392/4
https://github.com/kaloz/mwlwifi/commit/574e24e7dd661676e585dacd54e4f93b24e78ba0

Signed-off-by: Tad <tad@spotco.us>
---
 package/base-files/files/etc/init.d/amsdu-disable | 10 ++++++++++
 1 file changed, 10 insertions(+)
 create mode 100755 package/base-files/files/etc/init.d/amsdu-disable

diff --git a/package/base-files/files/etc/init.d/amsdu-disable b/package/base-files/files/etc/init.d/amsdu-disable
new file mode 100755
index 0000000000..a83dc32dd6
--- /dev/null
+++ b/package/base-files/files/etc/init.d/amsdu-disable
@@ -0,0 +1,10 @@
+#!/bin/sh /etc/rc.common
+
+USE_PROCD=0
+
+START=18
+STOP=30
+
+echo "0" >> /sys/kernel/debug/ieee80211/phy0/mwlwifi/tx_amsdu
+echo "0" >> /sys/kernel/debug/ieee80211/phy1/mwlwifi/tx_amsdu
+logger "AMSDU Disabled"
-- 
2.31.1

